sys/include/net/gnrc/lwmac/lwmac.h

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *               2016 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_lwmac Simplest possible MAC layer
+ * @ingroup     net_gnrc
+ * @brief       Lightweight MAC protocol that allows for duty cycling to save
+ *              energy.
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the LWMAC protocol
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ */
+
+#ifndef GNRC_LWMAC_H
+#define GNRC_LWMAC_H
+
+#include "kernel_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Time between consecutive wake-ups.
+ *
+ * This parameter governs power consumption, latency and throughput!
+ * In LWMAC, devices adopt duty-cycle scheme to conserve power. That is,
+ * time is divided into repeated cycles (or, superframes), and in each
+ * cycle, a node only wakes up for a period of time for receiving potential
+ * incoming packets for itself. This macro defines the wake-up interval, or,
+ * in other words, defines the cycle duration used in LWMAC. If the wake-up interval
+ * is short, nodes will wake up more frequently, which also increases
+ * the chances for receiving packets from neighbors (i.e., leads to higher
+ * throughput), but also results in higher power consumption.
+ * In LWMAC, by default, we regard the wake-up period as the beginning of a cycle.
+ */
+#ifndef LWMAC_WAKEUP_INTERVAL_US
+#define LWMAC_WAKEUP_INTERVAL_US        (100U * US_PER_MS)
+#endif
+
+/**
+ * @brief The Maximum WR (preamble) duration time.
+ *
+ * Since LWMAC adopts duty-cycle scheme, a node only wakes up for a short
+ * period in each cycle. Thus, to probe where is the wake-up period of the
+ * receiver, a sender sends WR (preamble) packets to notice the receiver for
+ * communication. To ensure that the receiver will catch at least one WR
+ * packet in one cycle, the sender repeatedly broadcasts a stream of WR packets
+ * with the broadcast duration (preamble duration) slightly longer period than
+ * LWMAC_WAKEUP_INTERVAL_US.
+ */
+#ifndef LWMAC_PREAMBLE_DURATION_US
+#define LWMAC_PREAMBLE_DURATION_US      ((13 * LWMAC_WAKEUP_INTERVAL_US) / 10)
```

Comment:
Yes. Otherwise, C will use "int" as default type for integer literals, and it uses the leftmost type as target type in a calculation. So "1 * <whatever" is implicitly converted to "int".
That causes trouble on 16bit platforms, e.g., "1 * US_PER_SEC" results in "1000000 % 2**16", which is usually not the desired effect. Thus, wherever numbers larger than 2**16 are expected, use "LU" as type suffix...